### PR TITLE
Added a comment field to bootstrap records.

### PR DESCRIPTION
**Pull request by tomcollinsproject**
Also, changed the bootstrap where_clause field from a varchar(255) to a text type.

Please note that this is a replacement for an earlier pull request from my 'bootstrap_comment' branch: https://github.com/zendesk/maxwell/pull/1373. Sorry for the confusion.

Reference: https://github.com/zendesk/maxwell/issues/1251

### CODE REVIEW

**Comment by tomcollinsproject** [comment]
Okay, I have things _somewhat_ sorted out now. Sorry for the second branch of this. There was some git clownery on my end 😕. 

A couple things to note: I could not get the integration tests to work on my machine without some help from a person on my team more familiar with Go. He was able to compile the `onetimeserver` for a 64 bit Mac OS and I cached that artifact locally. The 32 binary only works on OS X < 10.15.

Second, there are many tests that did not expect bootstrap table change events caused by my migration. I'm particularly concerned about a couple of the tests in `RecoveryTest` I'm not sure if my changes are okay or indicative of a larger problem. Any help or feedback is greatly appreciated.

Thanks.

File: src/test/java/com/zendesk/maxwell/recovery/RecoveryTest.java [hunk 1]
```diff
@@ -379,7 +379,8 @@ public void testSchemaIdRestore() throws Exception {
 		LOGGER.info("New pos: " + newPosition);
 		MysqlSavedSchema newSavedSchema = MysqlSavedSchema.restore(context, newPosition);
 		LOGGER.info("New schema id: " + newSavedSchema.getSchemaID());
-		assertEquals(new Long(oldSchemaId + 1), newSavedSchema.getSchemaID());
+		// TODO: this is oldSchemaId +3 insteand of +1 to offset for the maxwell.bootstrap schema migrations and does not seem like a viable long term solution
+		assertEquals(new Long(oldSchemaId + 3), newSavedSchema.getSchemaID());
```
Comment (osheroff) [inline src/test/java/com/zendesk/maxwell/recovery/RecoveryTest.java]: ah.  ok, sorry I wasn't quite clear before about the checked in SQL and the schema migrations; please keep *both* the changes to the SQL file and the code that upgrades old schemas.  So new schemas just get the in place schema but old ones are gracefully updated.  then this odd change to this test will go away.

**Comment by osheroff** [comment]
>  He was able to compile the onetimeserver for a 64 bit Mac OS and I cached that artifact locally. The 32 bit binary only works on OS X < 10.15.

Thanks for doing this / going through that.  Any chance of opening a PR?

**Comment by osheroff** [comment]
> I'm particularly concerned about a couple of the tests in RecoveryTest

I looked at the failures in Travis, those are normal for that goddamn test.  I've spent hours trying to stabilize that test but it tests far too many moving parts and it pigheadishly refuses to pass with great regularity.

**Comment by tomcollinsproject** [comment]
All the requests sound good to me. I'll try to get a PR for that Go onetimeserver as well. Thanks again!

**Comment by tomcollinsproject** [comment]
Okay, updated and only that one `RecoveryTest` is failing. A lot of the churn caused by updating the constructor has been backed out. As for the `onetimeserver`, my friend will submit a PR for that, but he was wondering: do you want to continue to support for 32 bit arch as well or just 64 bit going forward?

**Comment by tomcollinsproject** [comment]
Any thoughts on this? Is the code where it needs to be after this last check in?

**Comment by tomcollinsproject** [comment]
@osheroff, do you have any time to look at this?

**Comment by osheroff** [comment]
merged via #1404

**Pull request merged**